sgx-types/src/sig.rs

```diff
@@ -61,58 +61,88 @@ pub struct Author {
     reserved: [u32; 21],
 }
 
-/// The `Contents` of an enclave
-///
-/// This structure encompasses the second block of fields from `SIGSTRUCT`
-/// that is included in the signature. It is split out from `Signature`
-/// in order to make it easy to hash the fields for the signature.
-#[repr(C)]
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct Contents {
+/// The enclave parameters
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+pub struct Parameters {
     /// Bit vector specifying extended SSA frame feature set to be used.
     pub misc: Masked<MiscSelect>,
-    reserved0: [u8; 20],
+
     /// Enclave attributes struct.
     pub attr: Masked<Attributes>,
-    /// SHA256 hash of enclave contents.
-    pub mrenclave: [u8; 32],
-    reserved1: [u8; 32],
+
     /// User-defined value used in key derivation.
     pub isv_prod_id: isv::ProdId,
+
     /// User-defined value used in key derivation.
     pub isv_svn: isv::Svn,
 }
 
-impl From<[u8; 32]> for Contents {
-    fn from(value: [u8; 32]) -> Self {
-        Self::new(
-            Default::default(),
-            Default::default(),
-            value,
-            Default::default(),
-            Default::default(),
-        )
-    }
-}
-
-impl Contents {
-    /// Creates new SIGSTRUCT Contents from known values (including MRENCLAVE).
-    pub const fn new(
-        misc: Masked<MiscSelect>,
-        attr: Masked<Attributes>,
-        mrenclave: [u8; 32],
-        isv_prod_id: isv::ProdId,
-        isv_svn: isv::Svn,
-    ) -> Self {
-        Self {
-            misc,
+impl Parameters {
+    /// Combines the parameters and a hash of the enclave to produce a `Measurement`
+    pub const fn measure(&self, mrenclave: [u8; 32]) -> Measurement {
```

Comment:
Just a comment: calling this `measure(...)` is slightly confusing to me since the actual measurement `mrenclave` is an input and so the function is not really "measuring". I would have expected `measure(...)` to call something that produces `mrenclave` as output. Probably can ignore this for now, though.

Reply:
Honestly, I had the same thoughts. But I couldn't come up with a better name. Any suggestions?

Reply:
`measurement_wrapper`? `measurement_constructor`? `measurement_interface`? It's hard to think of something succinct. `include_measurement`? `add_mrenclave`?

Reply:
Though, we don't need to get stuck on this if all of those are unappealing.

Reply:
@lkatalin How about we leave it as it is for now and file a debt issue?

Reply:
Never mind. Fixed.